synedrion/src/paillier/encryption.rs

```diff
@@ -119,8 +108,9 @@ impl<P: PaillierParams> Ciphertext<P> {
         // Since `m` can be negative, we calculate `m * N ± 1` (never overflows since `m < N`),
         // then conditionally negate modulo N^2
 
-        let prod = abs_plaintext.mul_wide(pk.modulus());
-        let mut prod_mod = prod.to_montgomery(pk.monty_params_mod_n_squared());
+        let abs_plaintext = abs_plaintext.to_montgomery(pk.monty_params_mod_n_squared());
```

Comment:
This decreases performance somewhat, but makes it possible to use this method both for `Uint` and `WideUint`. Worth it?